lib/rspec-puppet/example/fact_example_group.rb

```diff
@@ -0,0 +1,56 @@
+module RSpec::Puppet
+  # This module provides support for custom facts
+  module FactExampleGroup
+    def subject
+      setup_facter do
+        Facter.fact(self.class.top_level_description)
+      end
+    end
+
+    # TODO: proper matcher with a description
+    def have_value(value)
+      have_attributes(value: value)
+    end
+
+    def rspec_puppet_cleanup
+      Facter.clear
+      # TODO: clean LOAD_PATH again?
+    end
+
+    private
+
+    # TODO: duplicates adapter
```

Comment:
I think `Rspec::Puppet::Support` should be split into a base adapter that also applies to facts and one that includes a catalog. Then you can call `setup_puppet` properly there. However, that's a larger effort I don't have time for now.